File: rest/src/main/java/io/sparkled/viewmodel/ModelConverter.java
package io.sparkled.viewmodel;

/**
 * Handles model -> view model conversion.
 *
 * @param <M>  A model class
 * @param <VM> A view model class
 */
public interface ModelConverter<M, VM extends ViewModel> {

    /**
     * @param model The model class to be converted
     * @return The converted view model.
     */
    VM toViewModel(M model);
}

File: rest/src/main/kotlin/io/sparkled/viewmodel/ModelConverter.kt
package io.sparkled.viewmodel

/**
 * Handles model -> view model conversion.

 * @param <M>  A model class
 * *
 * @param <VM> A view model class
</VM></M> */
interface ModelConverter<M, VM : ViewModel> {

    /**
     * @param model The model to be converted
     * *
     * @return The converted view model.
     */
    fun toViewModel(model: M): VM
}
